archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/expression/evaluator/ExpressionHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.archetype.engine.v2.expression.evaluator;
+
+/**
+ * Evaluate an expression.
+ *
+ * @param <T> Result type.
+ */
+public interface ExpressionHandler<T> {
+
+    /**
+     * Evaluate an expression and return the {@code Literal} as a result.
+     *
+     * @return Literal.
+     * @throws ParserException if a parsing error occurs.
+     */
+    Literal<T> evaluate() throws ParserException;
```

Comment:
Why? As for me it is natural that the result of the expression is a literal

Reply:
A literal is a fixed value in the source code. The result of an evaluation is a computed value, we cannot call that a literal.

You could model something more abstract like Value, and specialize it. Maybe we don't need to model literals but just constants and variables.

Reply:
I am not agree with you. A literal is the source code representation of a fixed value. It's possible to assign a literal to variable. As for me 'literal' and 'value' are the similar, so I do not see any reason to change the name of the class or create new hierarchy.

Reply:
See https://en.wikipedia.org/wiki/Literal_(computer_programming)

Building an AST is overkill for what we need (my experiment I gave as a reference was bad). Without an AST we can't argue on semantics :)

See this implementation as an example: 
 - https://github.com/javaee/openmq-mq4/blob/78fc3a5f0d0a0e77fa99e3c247ff6df8f1167c16/mq/src/share/java/com/sun/messaging/jmq/util/selector/Selector.java
 - https://github.com/javaee/openmq-mq4/blob/78fc3a5f0d0a0e77fa99e3c247ff6df8f1167c16/mq/src/share/java/com/sun/messaging/jmq/util/selector/SelectorToken.java

We also need to do an infix to RPN conversion, see this link as an example: https://www.andreinc.net/2010/10/05/converting-infix-to-rpn-shunting-yard-algorithm

Reply:
The goal of the task is `...implement a generic boolean expression evaluator with tests...`  The example were given in the spec and in the tests of your branch with your experiments. My code is working code and it solve the given task, so I do not understand why I need to change something, except maybe renaming . If we need the conversion from infix to RPN notation it can easily be done after parsing because we have AST. If we need that the parser can parse RPN notation it is other thing. For this I have to implement other algorithm that can parse such type of expressions.

Reply:
Fine. Let's merge it as is.